### PR TITLE
Variable support

### PR DESCRIPTION
**Pull request by plynchnlm**

### CODE REVIEW

File: src/fhirpath.js [hunk 1]
```diff
@@ -462,19 +470,31 @@ var parse = function(path) {
...
+  // Set up default standard variables, and allow override from the variables.
+  // However, we'll keep our own copy of dataRoot for internal processing.
+  let vars = {context: resource, ucum: 'http://unitsofmeasure.org'};
+  let ctx = {dataRoot, vars: Object.assign(vars, context)};
+  return engine.doEval(ctx, dataRoot, parsedPath.children[0]);
 }
 
 /**
+ *  Evaluates the "path" FHIRPath expression on the given resource, using data
+ *  from "context" for variables mention in expression.
```
Comment (akanduru) [inline src/fhirpath.js]: typo: mentioned

File: src/fhirpath.js [hunk 2]
```diff
@@ -462,19 +470,31 @@ var parse = function(path) {
 /**
  *  Applies the given parsed FHIRPath expression to the given resource,
  *  returning the result of doEval.
+ * @param {(object|object[])} resource -  FHIR resource, bundle as js object or array of resources
+ * @param {string} path - fhirpath expression, sample 'Patient.name.given'
```
Comment (akanduru) [inline src/fhirpath.js]: mismatched param name. parsedPath is not the same as path?

File: src/fhirpath.js (hunk 1 above)
Comment (plynchnlm) [reply src/fhirpath.js]: Fixed.

File: src/fhirpath.js (hunk 2 above)
Comment (plynchnlm) [reply src/fhirpath.js]: Fixed.

**Pull request merged**